packages/alchemy/src/middleware/simulate-uo.ts

```diff
@@ -0,0 +1,30 @@
+import {
+  deepHexlify,
+  resolveProperties,
+  type UserOperationStruct,
+} from "@alchemy/aa-core";
+import type { AlchemyProvider } from "../provider/base.js";
+import type { ClientWithAlchemyMethods } from "./client.js";
+
+export const withAlchemyUserOpSimulation = <P extends AlchemyProvider>(
+  provider: P
+): P => {
+  provider.withSimulateUOMiddleware(async (uoStruct) => {
+    const uoSimResult = await (
+      provider.rpcClient as ClientWithAlchemyMethods
+    ).request({
+      method: "alchemy_simulateUserOperationAssetChanges",
+      params: [
+        deepHexlify(await resolveProperties(uoStruct)) as UserOperationStruct,
```

Comment:
Would be great to confirm that we're sending a dummy signature not the real one

Reply:
confirmed: https://github.com/alchemyplatform/aa-sdk/blob/cd5715f9/packages/core/src/account/simple.ts#L32